x-pack/platform/plugins/shared/task_manager/server/saved_objects/schemas/rrule.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { Frequency } from '@kbn/rrule';
+import moment from 'moment-timezone';
+
+function validateTimezone(timezone: string) {
+  if (moment.tz.zone(timezone) != null) {
+    return;
+  }
+
+  return 'string is not a valid timezone: ' + timezone;
+}
+
+const validateRecurrenceByWeekday = (array: string[]) => {
+  if (array.length === 0) {
+    return 'rRule byweekday cannot be empty';
+  }
+
+  const byWeekDayRegex = new RegExp('^(((\\+|-)[1-4])?(MO|TU|WE|TH|FR|SA|SU))$');
+  const invalidDays: string[] = [];
+
+  array.forEach((day) => {
+    if (!byWeekDayRegex.test(day)) {
+      invalidDays.push(day);
+    }
+  });
+
+  if (invalidDays.length > 0) {
+    return `invalid byweekday values in rRule byweekday: ${invalidDays.join(',')}`;
+  }
+};
+
+const rruleCommon = schema.object({
+  freq: schema.maybe(
```

Comment:
did you want to remove these schema.maybes here or do another backport?

Reply:
Thanks for reminding, have just removed them.